examples/hello_world/client/nodejs/index.js

```diff
@@ -28,42 +29,19 @@ const SERVICE_PROTO_PATH = `${REPOSITORY_ROOT}/examples/hello_world/proto/hello_
 const OAK_LABEL_GRPC_METADATA_KEY = 'x-oak-label-bin';
 const OAK_SIGNED_CHALLENGE_GRPC_METADATA_KEY =
   'x-oak-signed-auth-challenge-bin';
-
-// Generated using the command:
-// ```shell
-// cargo run --manifest-path=oak_sign/Cargo.toml -- \
-//     generate \
-//     --private-key=http-test.key \
-//     --public-key=http-test.pub
-// ```
-// TODO(#1859): Generate a key in JS, instead of hardcoding one.
-const PUBLIC_KEY = Buffer.from(
-  'i7g/Mgft/2hsZBGRd+MF4f+Eir2Iyamkch8l4vPgLac=',
-  'base64'
-);
-
-// Generated using the command:
-// ```shell
-// cargo run --manifest-path=oak_sign/Cargo.toml -- \
-//     sign \
-//     --private-key=http-test.key \
-//     --input-string="oak-challenge" \
-//     --signature-file=http-test.sign
-// ```
-// TODO(#1859): Sign the hash in JS, instead of hardcoding this.
-const SIGNED_HASH = Buffer.from(
-  'ijbZOQ9mTtDEQyDEt+G3Rh0E0zwnY6+DkRd7bkllJINJbAhIiss3krdOzWo9d/iz5NJ9XGbp/qutOjAPSBkyDg==',
-  'base64'
-);
+const OAK_CHALLENGE = 'oak-challenge';
 
 async function main() {
-  const protos = await protobufjs.load([
-    OAK_LABEL_PROTO_PATH,
-    OAK_IDENTITY_PROTO_PATH,
-    SERVICE_PROTO_PATH,
+  const [protos, keyPair] = await Promise.all([
```

Comment:
I know absolutely nothing about JS... so I am curious, could not you leave the declaration of `protos`  as before, and add a separate declaration for `keyPair`? I guess it has something to do with `Promise.all`. What does it do? Why is it needed?

Reply:
I would imagine if you do one promise after the other it would work, but the second one would not start until the first one is resolved, this looks like it starts both in parallel, so it resolves when both of them resolve. (disclaimer: I also don't know much about JS, but I guess this is the same for any language that has async support, including Rust)

Reply:
Makes sense. Thanks.

Reply:
Yup, what @tiziano88 said :)